### PR TITLE
[cpu] Improve the ability to debug JITted code

### PR DESCRIPTION
**Pull request by opti-mix**
* Add the option --dump-jit-symbol-info to dump addresses of JITted functions
  This option can be very useful e.g. for debugging crashes inside the code produced by the CPU JIT.

* Add debugger events listener
   This allows to the JIT to inform the debugger about the dynamically generated object file. Using this information, the debugger can show proper stack traces, etc.
   This feature currently works only on Linux, because LLVM's MachO RuntimeDyld does not support debug info loading yet.

The main motivation for this PR was the debugging experience last week, when @beicy was hunting a bug in the JITted code. This PR should make it easier to debug the JITted code.

### CODE REVIEW

File: lib/Backends/CPU/GlowJIT.h [hunk 1]
```diff
@@ -46,6 +46,9 @@ class GlowJIT {
   const DataLayout DL_;
   RTDyldObjectLinkingLayer objectLayer_;
   IRCompileLayer<decltype(objectLayer_), SimpleCompiler> compileLayer_;
+  /// The listener for debugger events. It is used to provide debuggers with the
+  /// information about JITted code.
+  llvm::JITEventListener *dbgRegistrationListener = nullptr;
```
Comment (bertmaher) [inline lib/Backends/CPU/GlowJIT.h]: Could this be a member of `NotifyLoadedFunctor` instead?  Also, nit: end member name with _.

File: lib/Backends/CPU/GlowJIT.cpp [hunk 2]
```diff
@@ -15,14 +15,81 @@
...
+    }
+  }
+
+public:
+  NotifyLoadedFunctor(GlowJIT *jit) : jit_(jit) {}
+  void operator()(llvm::orc::RTDyldObjectLinkingLayerBase::ObjHandleT,
+                  const llvm::orc::RTDyldObjectLinkingLayerBase::ObjectPtr &obj,
+                  const llvm::RuntimeDyld::LoadedObjectInfo &objInfo) {
+    auto loadedObj = obj->getBinary();
+    // Inform the debugger about the loaded object file. This should allow e.g.
```
Comment (rdzhabarov) [inline lib/Backends/CPU/GlowJIT.cpp]: nit: can you fix the second sentence?

File: lib/Backends/CPU/GlowJIT.h (hunk 1 above)
Comment (opti-mix) [reply lib/Backends/CPU/GlowJIT.h]: Done

File: lib/Backends/CPU/GlowJIT.cpp (hunk 2 above)
Comment (opti-mix) [reply lib/Backends/CPU/GlowJIT.cpp]: Done.

**Pull request merged**